magda-web-client/src/Components/Common/ErrorMessageBox.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { FunctionComponent, useRef, useEffect } from "react";
+import { connect } from "react-redux";
+import { config } from "config";
+//import ServerError from "";
+import FileDeletionError from "helpers/FileDeletionError";
+import ServerError from "Components/Dataset/Add/Errors/ServerError";
+
+type PropsType = {
+    /**
+     * redux global state data
+     * use to access error that shared via redux state
+     * available via `react-redux`. Should not passing via component props
+     *
+     * @type {*}
+     */
+    state?: any;
+
+    /**
+     * use for get error object from redux state
+     *
+     * @type {((state: any) => Error | null)}
+     */
+    stateErrorGetter?: (state: any) => Error | null;
+
+    /**
+     * error object that should be passed via component props
+     * Only of `error` or `stateErrorGetter` should be specified
+     * if both present, `error` has higher priority
+     *
+     * @type {(Error | null | string)}
+     */
+    error?: Error | null | string;
+
+    /**
+     * optional; auto move viewport to this error message box; default to false
+     *
+     * @type {boolean}
+     */
+    scrollIntoView?: boolean;
+};
+
+const ErrorMessageBox: FunctionComponent<PropsType> = (props) => {
```

Comment:
It's now being used by a lot of other places (e.g. modals etc) and feels it's strange to pull it from `/add` folder.
Its functionality is quite generic e.g. you can set error either by props or redux.
You also can overwrite the error message using a customise Error (and getErrorContent)
I think we probably can gradually move all code to use this ErrorMessageBox (probably not now) to avoid some repetition job.